terraform/modules/arm-builder-vm/ubuntu-builder.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2024 Technology Innovation Institute (TII)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -x # debug
+
+################################################################################
+
+# Assume root if HOME and USER are unset
+[ -z "${HOME}" ] && export HOME="/root"
+[ -z "${USER}" ] && export USER="root"
+
+################################################################################
+
+apt_update() {
+	sudo apt-get update -y
+	sudo apt-get upgrade -y
+	sudo apt-get install -y ca-certificates curl xz-utils
+}
+
+install_nix() {
+	type="$1"
+	if [ "$type" = "single" ]; then
+		# Single-user
+		sh <(curl -L https://nixos.org/nix/install) --yes --no-daemon
+	elif [ "$type" = "multi" ]; then
+		# Multi-user
+		sh <(curl -L https://nixos.org/nix/install) --yes --daemon
+	else
+		echo "Error: unknown installation type: '$type'"
+		exit 1
+	fi
+	# Fix https://github.com/nix-community/home-manager/issues/3734:
+	sudo mkdir -m 0755 -p /nix/var/nix/{profiles,gcroots}/per-user/"$USER"
+	sudo chown -R "$USER:nixbld" "/nix/var/nix/profiles/per-user/$USER"
+	# Enable flakes
+	extra_nix_conf="experimental-features = nix-command flakes"
+	sudo sh -c "printf '$extra_nix_conf\n'>>/etc/nix/nix.conf"
+	# https://github.com/NixOS/nix/issues/1078#issuecomment-1019327751
+	for f in /nix/var/nix/profiles/default/bin/nix*; do
+		sudo ln -fs "$f" "/usr/bin/$(basename "$f")"
+	done
+}
+
+configure_builder() {
+	# Add user: nix
+	# Extra nix config for the builder,
+	# for detailed description of each of the below options see:
+	# https://nixos.org/manual/nix/stable/command-ref/conf-file
+	extra_nix_conf="
+# 20 GB (20*1024*1024*1024)
+min-free = 21474836480
+# 200 GB (200*1024*1024*1024)
+max-free = 214748364800
```

Comment:
Please make sure this is correct in your new aarch builders too.
For instance, the comment 'Add user: nix' is clearly not correct.
Also, the arm builder disk seems to be 100GB, so the max-free of 200GB is probably not correct.